src/App.tsx

```diff
@@ -1,26 +1,175 @@
+        setIsInputDisabled(false);
+        if (inputRef.current) {
+          inputRef.current?.focus();
+        }
+      });
+  }
+
+  function handleSubmit(event: React.FormEvent) {
+    event.preventDefault();
+    if (!newTodoTitle.trim()) {
+      setErrorMessage(Error.Empty);
+      setTimeout(() => setErrorMessage(''), 3000);
+
+      return;
+    }
+
+    const loadingTodo: Todo = {
+      userId: USER_ID,
+      title: newTodoTitle.trim(),
+      completed: false,
+      id: 0,
+    };
+
+    setTempTodo(loadingTodo);
+    setIsInputDisabled(true);
+    addPost(loadingTodo);
+  }
+
+  function deletePost(todoId: number) {
+    setTodoIdLoading(todoId);
+    deleteTodo(todoId)
+      .then(() => {
+        setTodos(currentTodos =>
+          currentTodos.filter(todo => todo.id !== todoId),
+        );
+        setTodoIdLoading(null);
+      })
+      .catch(() => {
+        setErrorMessage(Error.Delete);
+        setTodoIdLoading(null);
+        setTimeout(() => setErrorMessage(''), 3000);
+      })
+      .finally(() => {
+        if (!isInputDisabled && inputRef.current) {
+          inputRef.current?.focus();
+        }
+      });
+  }
+
+  function deleteCompleted() {
+    todos.map(todo => {
+      if (todo.completed) {
+        deletePost(todo.id);
+      }
+    });
+  }
+
+  const filteredTodos = useMemo(() => {
+    switch (filteredBy) {
+      case FilterName.ACTIVE:
+        return todos.filter(todo => !todo.completed);
+      case FilterName.COMPLETED:
+        return todos.filter(todo => todo.completed);
+      default:
+        return todos;
+    }
+  }, [todos, filteredBy]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          activeTodos={activeTodos}
+          handleSubmit={handleSubmit}
+          handleNewTodoTitleChange={handleNewTodoTitleChange}
+          isInputDisabled={isInputDisabled}
+          newTodoTitle={newTodoTitle}
+          inputRef={inputRef}
+        />
+        {todos.length > 0 && (
+          <TodoList
+            filteredTodos={filteredTodos}
+            tempTodo={tempTodo}
+            todoIdLoading={todoIdLoading}
+            deletePost={deletePost}
+          />
+        )}
+        {todos.length > 0 && (
+          <Footer
+            activeTodos={activeTodos}
+            filteredBy={filteredBy}
+            setFilteredBy={setFilteredBy}
+            todos={todos}
+            deleteCompleted={deleteCompleted}
+          />
```

Comment:
1) don't forget about indents
2) u can use `!!` to convert value to a boolean type

```suggestion
          inputRef={inputRef}
        />
        
        {!!todos.length && (
          <TodoList
            filteredTodos={filteredTodos}
            tempTodo={tempTodo}
            todoIdLoading={todoIdLoading}
            deletePost={deletePost}
          />
        )}
        
        {!!todos.length && (
          <Footer
            activeTodos={activeTodos}
            filteredBy={filteredBy}
            setFilteredBy={setFilteredBy}
            todos={todos}
            deleteCompleted={deleteCompleted}
          />
```